do_everything.sh

```diff
@@ -123,166 +114,7 @@ echo
-    # TODO: It seems like there may be a better way to handle the test issues
-    # http://stackoverflow.com/questions/22055741/googletest-for-android-ndk
-    # https://source.android.com/reference/com/android/tradefed/testtype/GTest.html
-    apply_patch $my_loc/patches/camera_info_manager.patch
-
-    # Patch camera_calibration_parsers - deleted python things and solved problem finding Boost.
-    apply_patch $my_loc/patches/camera_calibration_parsers.patch
-
-    # Patch cv_bridge - fix transitive linking in cv_bridge-extras.cmake
-    apply_patch $my_loc/patches/cv_bridge.patch
-
-    # Patch theora_image_transport - fix transitive linking
-    apply_patch $my_loc/patches/theora_image_transport.patch
-
-    # Patch robot_pose_ekf - Add bfl library cmake variables, also, remove tests
-    # TODO: The correct way to handle this would be to create .cmake files for bfl and do a findpackage(orocos-bfl)
-    # apply_patch $my_loc/patches/robot_pose_ekf.patch
-
-    # Patch robot_state_publisher - Add ARCHIVE DESTINATION
-    # TODO: Create PR to add ARCHIVE DESTINATION
-    apply_patch $my_loc/patches/robot_state_publisher.patch
-
-    # Patch moveit_core - Add fcl library cmake variables
-    # TODO: The correct way to handle this would be to create .cmake files for fcl and do a findpackage(fcl)
-    #apply_patch $my_loc/patches/moveit_core.patch
-
-    # Patch moveit_core plugins - Add ARCHIVE DESTINATION
-    # TODO: PR merged: https://github.com/ros-planning/moveit_core/pull/251
-    # Wait for next release to remove (current 0.6.15)
-    #apply_patch $my_loc/patches/moveit_core_plugins.patch
-
-    # Patch camera_calibration_parsers - Fix yaml-cpp dependency
-    # TODO: PR created: https://github.com/ros-perception/image_common/pull/36
-    # apply_patch $my_loc/patches/camera_calibration_parsers.patch
-
-    # Patch image_view - Solved YAML linking problems, and transitive linking.
-    apply_patch $my_loc/patches/image_view.patch
-
-    # Patch depth_image_proc - Solved transitive linking problems
-    apply_patch $my_loc/patches/depth_image_proc.patch
-
-    # Patch urdf - Fixed linking with pluginlib and dependencies in downstream packages
-    apply_patch $my_loc/patches/urdf.patch
-
-    # Patch global_planner - Add angles dependency
-    # TODO: PR merged: https://github.com/ros-planning/navigation/pull/359
-    # Wait for next release to remove (current 1.12.4)
-    # apply_patch $my_loc/patches/global_planner.patch
-
-    #Patch Poco lib
-    apply_patch $my_loc/patches/poco.patch
-
-    # Plugin specific patches
-    if [ $use_pluginlib -ne 0 ]; then
-        # Patch pluginlib for static loading
-        apply_patch $my_loc/patches/pluginlib.patch
-        # apply_patch image_transport # to fix faulty export plugins
-        apply_patch $my_loc/patches/image_transport.patch
-    fi
-
-    ## Demo Application specific patches
-
+    run_cmd apply_patches $my_loc/patches/source_patches $prefix
```

Comment:
Why the additional subdirectory `source_patches`? I would just leave them in `patches` and only move the unused ones to `patches/graveyard/` or delete them directly (they can still be found via the Git history).

Reply:
There's a single patch that has to be applied in a different directory: `android.toolchain.cmake.patch`; in principle that one has to be in a separate folder.
Perhaps we can move that one to `files` directory and move all the patches up one level as you propose?

Reply:
:+1:

An alternative approach I took in https://github.com/Intermodalics/ros_android_private/pull/17 is to commit a wrapper `android.toolchain.cmake` (or a template) and apply changes as CMake code after having included the original file, instead of patching the file in the Android NDK installation folder.

Such a custom and, if necessary, generated toolchain file could also replace most of the cmake options passed in [build_catkin_workspace.sh:103ff](https://github.com/Intermodalics/ros_android/blob/bcccc2127b8e8e4ccb7e99a2a0bc9739755b269d/build_catkin_workspace.sh#L103) and [utils.sh:21](https://github.com/Intermodalics/ros_android/blob/bcccc2127b8e8e4ccb7e99a2a0bc9739755b269d/utils.sh#L21) and be exposed to the user to be used in overlay workspaces (installed to target).